backend/de.metas.business/src/main/java/de/metas/costing/methods/AveragePOCostingMethodHandler.java

```diff
@@ -165,17 +165,12 @@ private CostDetailCreateResult createCostDetailAndAdjustCurrentCosts(final CostD
 		if (isInboundTrx || request.isReversal())
 		{
 			// Seed/initial costs import
-			if (request.getDocumentRef().isInventoryLine() && request.getQty().signum() == 0)
+			if (request.getDocumentRef().isInventoryLine() && request.isExplicitCostPrice())
 			{
 				requestEffective = request.withAmount(request.getAmt().toZero());
-				if (currentCosts.getCurrentQty().isZero())
-				{
+
```

Comment:
loud thinking: i feel like I would keep this check for now.
We have to think about the implications of removing it.

Reply:
Yes. I am changing things around there. Please, don't review yet :)

Reply:
i know, np